src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -367,13 +367,32 @@
            (throw (IllegalArgumentException. (format "Catalog must be specified as a string or a map, not '%s'" (class catalog)))))))
 
 (defmethod parse-catalog String
-  [catalog version]
+  [catalog version received-time]
   {:pre   [(string? catalog)]
    :post  [(map? %)]}
-  (parse-catalog (json/parse-string catalog true) version))
+  (parse-catalog (json/parse-string catalog true) version received-time))
+
+(defmethod parse-catalog 4
+  [catalog version received-time]
+  {:pre [(map? catalog)
+         (number? version)]
+   :post [(map? %)]}
+  (-> catalog
+      (assoc :producer_timestamp received-time)
```

Comment:
I think the invocation in question is https://github.com/puppetlabs/puppetdb/pull/1450/files#diff-d47aa6cc1d029cd621e847590c557d6cR28. It appears in this case that, if there's no producer_timestamp, we're using `(now)` to fill it out. We should instead use the `receiver_timestamp`. But that might be a misreading.